tiny_dnn/core/kernels/global_avepool_op_avx.h

```diff
@@ -0,0 +1,196 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/kernels/global_avepool_op_internal.h"
+
+namespace tiny_dnn {
+namespace kernels {
+
+#ifdef CNN_USE_AVX
+#ifndef CNN_USE_DOUBLE
+
+// float version
+inline void avx_global_avepool_forward_kernel(
+  const tensor_t &in_data,
+  tensor_t &out_data,
+  const core::global_avepool_params &params,
+  const bool layer_parallelize) {
+  const size_t pool_area            = params.in.width_ * params.in.height_;
+  const float pool_area_inv         = 1.0f / static_cast<float>(pool_area);
+  const size_t nblocks_per_channel  = pool_area / 8;
+  const size_t nremains_per_channel = pool_area & 7;
+
+  int32_t mask_src[] = {
+    -1, -1, -1, -1, -1, -1, -1, -1, 0, 0, 0, 0, 0, 0, 0, 0,
+  };
+  __m256i imask =
+    _mm256_loadu_si256((__m256i const *)(mask_src + 8 - nremains_per_channel));
+
+  for_i(layer_parallelize, in_data.size(), [&](int sample) {
```

Comment:
Please use `size_t sample` instead of `int sample` for_i functor lambda's argument.
#679 is related.